### DISCUSSION TITLE
How to inject trace baggage/tags into MDC

### DISCUSSION THREAD

**Discussion by moreirajo** [open-telemetry/opentelemetry-java-instrumentation#6241] [Q&A] [answered]
Hi,

I have a spring boot app using [spring cloud sleuth otel](https://github.com/spring-projects-experimental/spring-cloud-sleuth-otel) version `1.1.0-M6` and I want to add the trace baggage to MDC so my logs (sl4j) can print that key. 
It is worth noting that my app is reactive (webflux).

I tried:
```java
@Autowired private org.springframework.cloud.sleuth.Tracer tracer;
...
tracer.createBaggage("my-key", "some-value");
```
application.yaml:
```yam
spring:
  sleuth:
    baggage:
      correlation-fields:
      - my-key
```

It works when I create the baggage for a couple of logs but then stop working.

Am I missing something?

**Comment by mateuszrzeszutek** [comment]
Hey @moreirajo ,
Have you tried opening an issue in the Sleuth repo?

**Reply by moreirajo** [reply]
Hi, yes I did. Please anyone with input on this answer in https://github.com/spring-projects-experimental/spring-cloud-sleuth-otel/issues/116. Thanks.

**Comment by moreirajo** [comment, marked as answer]
This is more related with Sleuth.
Please anyone with input on this answer in https://github.com/spring-projects-experimental/spring-cloud-sleuth-otel/issues/116. 
Thanks.
